docs/conditions.rst

```diff
@@ -249,3 +251,172 @@ do not need actual record by are only looking to traverse::
+	$condition4 = new BasicCondition('surname', 'XYZ');
+	
+   // $compoundCondition2 is created using OR as junction and $condition3 and $condition4 as nested conditions
+	$compoundCondition2 = CompoundCondition::mergeOr($condition3, $condition4);
+
+	$condition5 = new BasicCondition('name', 'like', 'CDE%');
+	
+   // $compoundCondition3 is created using AND as junction and $condition5 and $compoundCondition2 as nested conditions
+	$compoundCondition3 = CompoundCondition::mergeAnd($condition5, $compoundCondition2);
+
+   // $compoundCondition is created using OR as junction and $compoundCondition1 and $compoundCondition3 as nested conditions
+	$compoundCondition = CompoundCondition::mergeOr($compoundCondition1, $compoundCondition3);
+	
+	
+CompoundCondition is an independent object not related to any model. Applying scope to model is using the Model::scope()->add($condition) method::
+
+	$contact->scope()->add($condition); // adding condition to a model
+	$contact->scope()->add($conditionXYZ); // adding more conditions
+	
+.. php:method:: __construct($nestedConditions = [], $junction = CompoundCondition::AND);
+
+Creates a CompoundCondition object from an array::
+
+	// below will create 2 conditions and nest them in a compound conditions with AND junction
+	$compoundCondition1 = new CompoundCondition([
+		['name', 'like', 'ABC%'],
+		['country', 'US']
+	]);
+	
+.. php:method:: negate();
+
+Negate method has behind the full map of conditions so any condition object can be negated, e.g negating '>=' results in '<', etc.
+For compound conditionss this method is using De Morgan's laws, e.g::
+
+	// using $compoundCondition1 defined above
+	// results in "(Name not like 'ABC%') or (Country does not equal 'US')"
+	$compoundCondition1->negate();
+
+.. php:method:: mergeAnd(AbstractCondition $conditionA, AbstractCondition $conditionB, $_ = null);
+
+Merge number of conditions using AND as junction. Returns the resulting CompoundCondition object.
+
+.. php:method:: mergeOr(AbstractCondition $conditionA, AbstractCondition $conditionB, $_ = null);
+
+Merge number of conditions using OR as junction. Returns the resulting CompoundCondition object.
+
+.. php:method:: simplify();
+
+Peels off single nested conditions. Useful for converting (((field = value))) to field = value.
+
+.. php:method:: clear();
+
+Clears the condition from nested conditions.
+
+.. php:method:: isOr();
+
+Checks if scope components are joined by OR
+
+.. php:method:: isAnd();
+
+Checks if scope components are joined by AND
+
+Conditions on Referenced Models
+-------------------------------
+
+Agile Data allows for adding conditions on related models for retrieval of type 'model has references where'.
+
+Setting conditions on references can be done utilizing the Model::refLink method but there is a shorthand format 
+directly integrated with addCondition method using "/" to chain the reference names::
+
+	$contact->addCondition('company/country', 'US');
```

Comment:
inspired with `/` by FS, or any popular lib? I would intuitively use `.`.

Reply:
This was the initial idea but it clashes with possibility to use . in field names as well.

Reply:
I am quite strongly for `.` as it is used normally in other libs.

Reply:
Then we have to exclude it as possible char for field name and I need @DarkSide666 review on this. We had discussions on it previously.

Reply:
I like `/` better than `.` because:
- dot is used in `table.field` syntax in `join()`
- theoretically dot can be used in field names
- theoretically dot can be used in `database.table.field` syntax too somewhere or `sequence.currval()` or things like these.

So my opinion is that dot should be used for things like _properties_, but slash used for _navigating_. That way we should never mix them accidentally.

Reply:
>     * dot is used in `table.field` syntax in `join()`

that is what I propose :) - the dot is "direct member operator"

>     * theoretically dot can be used in field names `last.name`

naming column like that is not supported by SQL nor DQL (without using backticks/escaping)

>     * theoretically dot can be used in `database.table.field` syntax too somewhere

same as above

> or `sequence.currval()` or things like these.

"direct member operator" can have 1:1 relationship

> 
> So my opinion is that dot should be used for things like _properties_ or _alias.field_, but slash used for _navigating_. That way we should never mix them accidentally.

it is about how column can be named, because `table.field` is already an navigation, but only 1:1

Reply:
I get the point - atk is not ready for this change now, but one very good news - `/` can be very easily deprecated in the future. So let's keep `/` for traversing strictly only.

Reply:
> naming column like that is not supported by SQL nor DQL (without using backticks/escaping)

Toolkit automatically always adds escaping so it's _supported_ (but not well tested I guess).

BTW, if we use dot instead of slash, then this doesn't look understandable at all?
`$contact->addCondition('company.tickets.#', '>', 3);`

Reply:
It does, because dot is scope is "direct member operator", ie. member can be table, field, custom/magic/internal function...

Reply:
And in some places it even soft-escapes `table.field` to `"table"."field"` so dot is used in lot of other places in toolkit and using it here can be confusing and risky in some situations.

Reply:
We was discussing this with @mvorisek for about hour and came to conclusion that:
- *should use slash here*